spec/support/api_spec_helper.rb

```diff
@@ -74,7 +74,7 @@ def api_config(param)
 
   def define_user
     @role  = FactoryGirl.create(:miq_user_role, :name => api_config(:role_name))
-    @group = FactoryGirl.create(:miq_group, :description => api_config(:group_name), :miq_user_role_id => @role.id)
+    @group = FactoryGirl.create(:miq_group, :description => api_config(:group_name), :miq_user_role => @role)
```

Comment:
I agree, the "before" way bypasses the normal interfaces and makes for subtle test bugs.